src/k8s/pkg/k8sd/features/contour/gateway_test.go

```diff
@@ -0,0 +1,204 @@
+		}
+
+		status, err := contour.ApplyGateway(context.Background(), snapM, gateway, network, nil)
+
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(status.Enabled).To(BeFalse())
+		g.Expect(status.Version).To(Equal(contour.ContourGatewayProvisionerContourImageTag))
+		g.Expect(status.Message).To(ContainSubstring("disabled"))
+		g.Expect(helmM.ApplyCalledWith).To(HaveLen(1))
+
+	})
+}
+
+func TestGatewayEnabled(t *testing.T) {
+	t.Run("HelmFailed", func(t *testing.T) {
+		g := NewWithT(t)
+
+		applyErr := errors.New("failed to apply")
+		helmM := &helmmock.Mock{
+			ApplyErr: applyErr,
+		}
+		snapM := &snapmock.Snap{
+			Mock: snapmock.Mock{
+				HelmClient: helmM,
+			},
+		}
+		network := types.Network{}
+		gateway := types.Gateway{
+			Enabled: ptr.To(true),
+		}
+
+		status, err := contour.ApplyGateway(context.Background(), snapM, gateway, network, nil)
+
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(err.Error()).To(ContainSubstring(applyErr.Error()))
+		g.Expect(status.Enabled).To(BeFalse())
+		g.Expect(status.Version).To(Equal(contour.ContourGatewayProvisionerContourImageTag))
+		g.Expect(status.Message).To(ContainSubstring("Failed to deploy Contour Gateway"))
+		g.Expect(helmM.ApplyCalledWith).To(HaveLen(1))
+	})
+
+	t.Run("Success", func(t *testing.T) {
+		g := NewWithT(t)
+
+		helmM := &helmmock.Mock{
+			ApplyChanged: true,
+		}
+		clientset := fake.NewSimpleClientset()
+		fakeDiscovery, ok := clientset.Discovery().(*fakediscovery.FakeDiscovery)
+		g.Expect(ok).To(BeTrue())
+		fakeDiscovery.Resources = []*v1.APIResourceList{
+			{
+				GroupVersion: "projectcontour.io/v1alpha1",
+				APIResources: []v1.APIResource{
+					{Name: "contourconfigurations"},
+					{Name: "contourdeployments"},
+					{Name: "extensionservices"},
+				},
+			},
+			{
+				GroupVersion: "projectcontour.io/v1",
+				APIResources: []v1.APIResource{
+					{Name: "tlscertificatedelegations"},
+					{Name: "httpproxies"},
+				},
+			}}
+		snapM := &snapmock.Snap{
+			Mock: snapmock.Mock{
+				HelmClient: helmM,
+				KubernetesClient: &kubernetes.Client{
+					Interface: clientset,
+				},
+			},
+		}
+		network := types.Network{}
+		gateway := types.Gateway{
+			Enabled: ptr.To(true),
+		}
+
+		status, err := contour.ApplyGateway(context.Background(), snapM, gateway, network, nil)
+
+		g.Expect(err).NotTo(HaveOccurred())
+		g.Expect(status.Enabled).To(BeTrue())
+		g.Expect(status.Version).To(Equal(contour.ContourGatewayProvisionerContourImageTag))
+		g.Expect(status.Message).To(ContainSubstring("enabled"))
+		g.Expect(helmM.ApplyCalledWith).To(HaveLen(2))
+
+		values := helmM.ApplyCalledWith[1].Values
+		projectcontour := values["projectcontour"].(map[string]any)["image"].(map[string]any)
+		g.Expect(projectcontour["repository"]).To(Equal(contour.ContourGatewayProvisionerContourImageRepo))
+		g.Expect(projectcontour["tag"]).To(Equal(contour.ContourGatewayProvisionerContourImageTag))
+		envoy := values["envoyproxy"].(map[string]any)["image"].(map[string]any)
```

Comment:
```suggestion
		envoyImage := values["envoyproxy"].(map[string]any)["image"].(map[string]any)
```
Also we should check for `envoyproxy` and `image` to exist first, otherwise we'll do a nil cast/reference.